### PR TITLE
feature: connected chains

### PR DESCRIPTION
**Pull request by MarkNerdi**
## Summary

> Please summarize your changes, describing **what** they are and **why** they were made.

...

## Changelog

```
- Please write a list of changes
```

### Platforms

> Please select any platforms where your changes have been tested.

-   **Desktop**
    -   [ ] MacOS
    -   [x] Linux
    -   [ ] Windows
-   **Mobile**
    -   [ ] iOS
    -   [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to **test and verify** that your changes work as intended.

...

## Checklist

> Please tick the following boxes that are relevant to your changes.

-   [ ] I have followed the contribution guidelines for this project
-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have added or modified tests that prove my changes work as intended
-   [ ] I have verified that new and existing unit tests pass locally with my changes
-   [ ] I have verified that my latest changes pass CI workflows for testing and linting
-   [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/desktop/components/drawers/network-config/drawers/ConnectedChainsDrawer.svelte
Comment (maxwellmattryan) [inline packages/desktop/components/drawers/network-config/drawers/ConnectedChainsDrawer.svelte:21]: I believe we could use the `$networkStatus` store for this right?

File: packages/desktop/components/drawers/network-config/drawers/components/index.ts
Comment (maxwellmattryan) [inline packages/desktop/components/drawers/network-config/drawers/components/index.ts:2]: I think this should be a shared component, within the `shared/components/atoms/pills` folder.

File: packages/desktop/components/drawers/network-config/drawers/components/index.ts
Comment (maxwellmattryan) [inline packages/desktop/components/drawers/network-config/drawers/components/index.ts:1]: For now we can put this in the `desktop/components` folder since it's not technically a drawer (unlikely to be used elsewhere, but still just for the sake of organization).

**Review by maxwellmattryan — CHANGES_REQUESTED** [review]
Have some minor requested changes :pray:

File: packages/desktop/components/drawers/network-config/drawers/components/NetworkCard.svelte
Comment (maxwellmattryan) [inline packages/desktop/components/drawers/network-config/drawers/components/NetworkCard.svelte:38]: We need to use the monospace font here rather than the regular one (not sure which one it is :sweat_smile:)

**Review by maxwellmattryan — CHANGES_REQUESTED** [review]

File: packages/desktop/components/drawers/network-config/drawers/ConnectedChainsDrawer.svelte
Comment (maxwellmattryan) [inline packages/desktop/components/drawers/network-config/drawers/ConnectedChainsDrawer.svelte:23]: This could be a util function i.e. `buildChainFromNetwork` or something

File: packages/shared/locales/en.json
Comment (maxwellmattryan) [inline packages/shared/locales/en.json:2291]: Typo

File: packages/shared/components/atoms/pills/index.js
Comment (maxwellmattryan) [inline packages/shared/components/atoms/pills/index.js:5]: Alphabetical order

**Review by maxwellmattryan — CHANGES_REQUESTED** [review]
Minor changes :pray:

File: packages/desktop/components/NetworkCard.svelte
Comment (maxwellmattryan) [inline packages/desktop/components/NetworkCard.svelte:40]: We need to change the font to be `code` font (forgot the name)

File: packages/desktop/components/NetworkCard.svelte
Comment (maxwellmattryan) [inline packages/desktop/components/NetworkCard.svelte:26]: When I hover over the hard, there's a tiny white outline around this icon :thinking:

File: packages/desktop/components/NetworkCard.svelte
Comment (maxwellmattryan) [reply packages/desktop/components/NetworkCard.svelte:40]: "My address" should be regular font, then the address itself needs to be `pre` - sorry if I got that mixed up in an earlier review comment :sweat_smile:

**Review by maxwellmattryan — CHANGES_REQUESTED** [review]

File: packages/shared/lib/core/network/utils/buildChainFromNetwork.ts
Comment (maxwellmattryan) [inline packages/shared/lib/core/network/utils/buildChainFromNetwork.ts:7]: Hmm we have two options here:
1. Keep as a util but pass parameters so that it doesn't read from a store
2. Turn into an action function 

I think for now 2 is easiest, although this has a "util" feel to me if that makes sense. 1 would mean we pass the stores in from the component, which may be nice for reactivity (e.g. if we switch an account), but I guess to access the account switcher the side drawer has to close anyway... :thinking:

File: packages/desktop/features/network.features.ts
Comment (maxwellmattryan) [inline packages/desktop/features/network.features.ts:9]: Need to change before merge

File: packages/shared/lib/core/network/interfaces/connected-chain.interface.ts
Comment (maxwellmattryan) [inline packages/shared/lib/core/network/interfaces/connected-chain.interface.ts:3]: Should be `IConnectedChain`, but this interface might go away anyway one we add the classes in b/c we'll have the metadata available from that then also query the status in real time from the class. We can keep this for now (still rename it though pls :pray:)

File: packages/shared/lib/core/network/interfaces/index.ts
Comment (maxwellmattryan) [inline packages/shared/lib/core/network/interfaces/index.ts:3]: Alphabetical order

File: packages/shared/lib/core/network/utils/buildChainFromNetwork.ts
Comment (maxwellmattryan) [inline packages/shared/lib/core/network/utils/buildChainFromNetwork.ts:2]: Might be good to import from `@core/profile/stores` to help out with weird circular deps later on

File: packages/shared/lib/core/network/utils/index.ts
Comment (maxwellmattryan) [inline packages/shared/lib/core/network/utils/index.ts:2]: Alphabetical order

**Review by maxwellmattryan — CHANGES_REQUESTED** [review]
Minor requested changes plus...

We don't currently have dark mode designs for this yet, but not sure if the card is suppose to be outlined. Feels a bit too much IMO (would be good to also ask @nicole-obrien):
Screenshot from 2023-04-21 14-39-20

File: packages/shared/lib/core/network/utils/buildChainFromNetwork.ts
Comment (MarkNerdi) [reply packages/shared/lib/core/network/utils/buildChainFromNetwork.ts:7]: this is only an mvp so that we can create the list of chains. In another step (maybe once we can actually add iscp chains) this has to be generalized such that we can build a persisted store.

for now i will go for 1, because like this its a bit of a blackbox, which i think is sometimes harder to understand

**Comment by nicole-obrien** [comment]
> Minor requested changes plus...
> 
> We don't currently have dark mode designs for this yet, but not sure if the card is suppose to be outlined. Feels a bit too much IMO (would be good to also ask @nicole-obrien): Screenshot from 2023-04-21 14-39-20

We don't have dark mode design yet so I wouldn't let that hold up this pr

**Review by maxwellmattryan — APPROVED** [review]
LGTM :fire:

**Pull request merged**
